File: OrienGO-Backend/src/main/java/com/example/oriengo/model/dto/TestResultResponseDTO.java
package com.example.oriengo.model.dto;

import com.example.oriengo.model.enumeration.Category;
import lombok.*;

import java.util.Map;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class TestResultResponseDTO {

    private Long id;

    private Long testId; // on expose juste l'id du Test, pas l'objet complet

    private Category dominantType;

    private String dominantTypeDescription;

    private Map<Category, Double> scores;

    private String keyPoints;

    private Long pdfId; // id du media PDF, s'il existe

    private boolean shared;

    private boolean downloaded;

    private boolean softDeleted;
}

File: OrienGO-Frontend/src/app/model/dto/TestResultResponseDTO.ts
import { Category } from "../enum/Category";

export interface TestResultResponseDTO {
  id: number;
  testId: number;
  dominantType: Category;
  dominantTypeDescription: string;
  scores: { [category in Category]?: number }; // Map<Category, Double> → Record<Category, number>
  keyPoints: string;
  pdfId?: number; // optional if it may be null in Java
  shared: boolean;
  downloaded: boolean;
  softDeleted: boolean;
}
